tsdb/ooo_head_read.go

```diff
@@ -253,6 +253,10 @@ func (cr *HeadAndOOOChunkReader) ChunkOrIterable(meta chunks.Meta) (chunkenc.Chu
 	}
 
 	s.Lock()
+	if !isOOO && meta.Chunk == nil && s.ooo == nil { // Don't need to merge if no OOO chunks.
```

Comment:
I'm trying to understand the performance implication of `s.ooo == nil`. It seems reasonable to have that condition, but it means that we'll try the merge more often than before. Does it make it better/same/worse than before #14396 ?

Reply:
I don't think #14396 changed performance much, and didn't change merging at all.
The big difference in performance and the way merges happen is in #14354.